mbed/SPI.h

```diff
@@ -101,41 +100,70 @@ class SPI {
     virtual int write(int value);
 
 #if DEVICE_SPI_ASYNCH
-    /** SPI transfer callback
-     *  @param Buffer the tx buffer
-     *  @param Buffer the rx buffer
-     *  @param int the event that triggered the calback
-     */
-    typedef mbed::util::FunctionPointer3<void, Buffer, Buffer, int> event_callback_t;
-
-    /** Start non-blocking SPI transfer.
-     *
-     * @param tx_buffer The TX buffer with data to be transfered. If NULL is passed,
-     *                  the default SPI value is sent
-     * @param tx_length The length of TX buffer
-     * @param rx_buffer The RX buffer which is used for received data. If NULL is passed,
-     *                  received data are ignored
-     * @param rx_length The length of RX buffer
-     * @param callback  The event callback function
-     * @param event     The logical OR of SPI events to modify. Look at spi hal header file for SPI events.
-     * @return Zero if the transfer has started, or -1 if SPI peripheral is busy
-     */
-    int transfer(void *tx_buffer, int tx_length, void *rx_buffer, int rx_length, const event_callback_t& callback, int event = SPI_EVENT_COMPLETE);
-
-     /** Start non-blocking SPI transfer.
-     *
-     * @param tx_buffer The TX buffer with data to be transfered. If NULL is passed,
-     *                  the default SPI Avalue is sent
-     * @param rx_buffer The RX buffer which is used for received data. If NULL is passed,
-     *                  received data are ignored
-     * @param callback  The event callback function
-     * @param event     The logical OR of SPI events to modify. Look at spi hal header file for SPI events.
-     * @return Zero if the transfer has started, or -1 if SPI peripheral is busy
+    class SPITransferAdder {
+        friend SPI;
+    private:
+        SPITransferAdder(SPI * owner);
+        const SPITransferAdder & operator =(const SPITransferAdder & a);
+        SPITransferAdder(const SPITransferAdder & a);
+    public:
+        /** Set the transmit buffer
+         *  Sets the transmit buffer pointer and transmit size.
+         *
+         *  NOTE: Repeated calls to tx() override buffer parameters.
+         *
+         *  @param[in] txBuf a pointer to the transmit buffer
+         *  @param[in] txSize the size of the transmit buffer
+         *  @return a reference to the SPITransferAdder
+         */
+        SPITransferAdder & tx(void * txBuf, size_t txSize);
+        /** Set the receive buffer
+         *  Sets the receive buffer pointer and receive size
+         *
+         *  NOTE: Repeated calls to rx() override buffer parameters.
+         *
+         *  @param[in] rxBuf a pointer to the receive buffer
+         *  @param[in] rxSize the size of the receive buffer
+         *  @return a reference to the SPITransferAdder
+         */
+        SPITransferAdder & rx(void * rxBuf, size_t rxSize);
```

Comment:
Can you please format the new code to be consistent with the current implementation?
For note we can use '@note'